### PR TITLE
Add componentWillAppear event

### PR DESCRIPTION
**Pull request by yogevbd**
Add `componentWillAppear` event which will notify the component that its view is about to be added to a view hierarchy.
This is equivalent to iOS [viewWillAppear](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621510-viewwillappear) event.

### CODE REVIEW

**Comment by guyca** [comment]
@yogevbd looks good! We should emit willAppear for external components as well.

**Comment by a-eid** [comment]
is this PR still being considered, is there any plans to support android as well ?

**Comment by yogevbd** [comment]
@a-eid yeah, we do plan on adding `componentWillAppear` for Android as well

File: lib/android/app/src/main/java/com/reactnativenavigation/views/component/ComponentLayout.java [hunk 1]
```diff
@@ -50,13 +51,20 @@ public void start() {
         reactView.start();
     }
 
-	public void sendComponentStart() {
-		reactView.sendComponentStart(ComponentType.Component);
-	}
+    public void sendComponentWillStart() {
+        if (!willAppearSent)
```
Comment (yogevbd) [inline lib/android/app/src/main/java/com/reactnativenavigation/views/component/ComponentLayout.java:55]: `willAppear` should be sent each time the component is going to be appeared as `didAppear` does

**Comment by nelsonprsousa** [comment]
@jpgarcia maybe this is something that could be added to rnn-hooks?

**Pull request merged**